packages/firestore/src/core/target_id_generator.ts

```diff
@@ -15,83 +15,43 @@
  * limitations under the License.
  */
 
-import { assert } from '../util/assert';
 import { TargetId } from './types';
 
-const RESERVED_BITS = 1;
-
-const enum GeneratorIds {
-  QueryCache = 0, // The target IDs for user-issued queries are even (end in 0).
-  SyncEngine = 1 // The target IDs for limbo detection are odd (end in 1).
-}
+/** Offset to ensure non-overlapping target ids. */
+const OFFSET = 2;
 
 /**
  * Generates monotonically increasing target IDs for sending targets to the
  * watch stream.
  *
- * The client constructs two generators, one for the query cache (via
- * forQueryCache()), and one for limbo documents (via forSyncEngine()). These
- * two generators produce non-overlapping IDs (by using even and odd IDs
+ * The client constructs two generators, one for the target cache, and one for
+ * for the sync engine (to generate limbo documents targets). These
+ * generators produce non-overlapping IDs (by using even and odd IDs
  * respectively).
  *
  * By separating the target ID space, the query cache can generate target IDs
  * that persist across client restarts, while sync engine can independently
  * generate in-memory target IDs that are transient and can be reused after a
  * restart.
  */
-// TODO(mrschmidt): Explore removing this class in favor of generating these IDs
-// directly in SyncEngine and LocalStore.
 export class TargetIdGenerator {
-  // Initialized in the constructor via call to seek().
-  private nextId!: TargetId;
-
-  /**
-   * Instantiates a new TargetIdGenerator. If a seed is provided, the generator
-   * will use the seed value as the next target ID.
-   */
-  constructor(private generatorId: number, seed?: number) {
-    assert(
-      (generatorId & RESERVED_BITS) === generatorId,
-      `Generator ID ${generatorId} contains more than ${RESERVED_BITS} reserved bits`
-    );
-    this.seek(seed !== undefined ? seed : this.generatorId);
-  }
+  constructor(private lastId: number) {}
 
   next(): TargetId {
-    const nextId = this.nextId;
-    this.nextId += 1 << RESERVED_BITS;
-    return nextId;
-  }
-
-  /**
-   * Returns the ID that follows the given ID. Subsequent calls to `next()`
-   * use the newly returned target ID as their base.
-   */
-  // PORTING NOTE: Multi-tab only.
-  after(targetId: TargetId): TargetId {
-    this.seek(targetId + (1 << RESERVED_BITS));
-    return this.next();
-  }
-
-  private seek(targetId: TargetId): void {
-    assert(
-      (targetId & RESERVED_BITS) === this.generatorId,
-      'Cannot supply target ID from different generator ID'
-    );
-    this.nextId = targetId;
+    this.lastId += OFFSET;
```

Comment:
This is functionally different from what came before. This is now a pre-increment where before it was post-increment. Just double-checking: is this intended?

Reply:
Yes. The target ID generator now gets initialized with 0 and -1 (shudder), which works around this inconsistency.